test/SimpleWebRequest.spec.ts

```diff
@@ -161,154 +172,199 @@ describe('SimpleWebRequest', () => {
+                blockResolver();
 
-            jasmine.Ajax.requests.mostRecent().respondWith({ status: 200 });
-            expect(onSuccessHigh2).toHaveBeenCalled();
+                jasmine.Ajax.requests.mostRecent().respondWith({ status: 200 });
 
-            // check if the request at the top of the queue got called again
-            expect(blockSpy).toHaveBeenCalledTimes(2);
+                return p3;
+            }).then(() => {
+                expect(onSuccessHigh2).toHaveBeenCalled();
 
-            jasmine.Ajax.requests.mostRecent().respondWith({ status: 200 });
-            expect(onSuccessCritical).toHaveBeenCalled();
+                jasmine.Ajax.requests.mostRecent().respondWith({ status: 200 });
+
+                return p1;
+            }).then(() => {
+                expect(onSuccessCritical).toHaveBeenCalled();
+            });
         });
 
         it('fails the request, if the blocking promise rejects', done => {
             SimpleWebRequestOptions.MaxSimultaneousRequests = 1;
             const url = faker.internet.url();
             const method = 'GET';
-            const blockDefer = SyncTasks.Defer<void>();
+            let blockRejecter: (err: any) => void = () => undefined;
+            const blockPromise = new Promise<void>((res, rej) => { blockRejecter = rej; });
             const errorString = 'Terrible error';
-            new SimpleWebRequest<string>(url, method, { priority: WebRequestPriority.Critical }, undefined, () => blockDefer.promise())
+            new SimpleWebRequest<string>(method, url, { priority: WebRequestPriority.Critical }, undefined, () => blockPromise)
                 .start()
                 .then(() => fail(), (err: WebErrorResponse) => {
                     expect(err.statusCode).toBe(0);
                     expect(err.statusText).toBe('_blockRequestUntil rejected: ' + errorString);
                     done();
                 });
 
-            blockDefer.reject(errorString);
+            blockRejecter(errorString);
         });
 
         it('does not attempt to fire aborted request, if it was aborted while blocked', () => {
             SimpleWebRequestOptions.MaxSimultaneousRequests = 1;
             const url = faker.internet.url();
             const method = 'GET';
-            const blockDefer = SyncTasks.Defer<void>();
-            new SimpleWebRequest<string>(url, method, { priority: WebRequestPriority.Critical }, undefined, () => blockDefer.promise())
-                .start()
-                .cancel();
+            let blockResolver: () => void = () => undefined;
+            const blockPromise = new Promise<void>((res, rej) => { blockResolver = res; });
+            const req = new SimpleWebRequest<string>(method, url, { priority: WebRequestPriority.Critical }, undefined, () => blockPromise);
+            req.start();
+            req.abort();
 
-            blockDefer.resolve(void 0);
+            blockResolver();
```

Comment:
Not using `done` or `return` but test is (or can be) async now, right? I worry if `count toBe 0` is always true here and the test is meaningless.